src/popup/pages/SignTransaction.vue

```diff
@@ -51,8 +58,24 @@ export default defineComponent({
           return;
         }
 
+        const replaceCallerTx = async (tx: Encoded.Transaction) => {
+          const unpackedTx = unpackTx(tx);
+          if (unpackedTx.tag === Tag.ContractCallTx || unpackedTx.tag === Tag.ContractCreateTx) {
+            // TxUnpacked isn't exported
+            // typescript is unaware that ContractCallTx and ContractCreateTx do have a callerId
+            (unpackedTx as any & { callerId: Encoded.AccountAddress })
+              .callerId = activeAccount?.address as Encoded.AccountAddress;
+            unpackedTx.nonce = (await aeSdk.api.getAccountByPubkey(
+              activeAccount?.address as Encoded.AccountAddress,
+            )).nonce + 1;
+          }
+          return buildTx(unpackedTx);
+        };
+
+        const txToSign = replaceCaller === 'true' ? await replaceCallerTx(transaction as Encoded.Transaction) : transaction;
```

Comment:
I think user should be aware of what is happening under the hood, and accept or decline such thing.

Reply:
how would we do this?